fbw-a32nx/src/systems/fmgc/src/flightplanning/ManagedFlightPlan.ts

```diff
@@ -1625,4 +1680,238 @@ export class ManagedFlightPlan {
+     * Missed approach engine outacceleration altitude for the origin airport
+     * @returns altitude in feet MSL, or undefined if none
+     */
+    public get missedEngineOutAccelerationAltitude(): number | undefined {
+        return ManagedFlightPlan.round(this.missedEngineOutAccelerationAltitudePilot ?? this.missedEngineOutAccelerationAltitudeDb, 10);
+    }
+
+    /**
+     * Check if the missed approach engine out acceleration altitude is derived from a pilot entry
+     * @returns true if pilot entry
+     */
+    public get isMissedEngineOutAccelerationAltitudePilotEntered(): boolean {
+        return this.missedEngineOutAccelerationAltitudePilot !== undefined;
+    }
+
+    /**
+     * Check if a default missed approach engine out acceleration altitude is available
+     * @returns true if default is available
+     */
+    public get hasMissedEngineOutAccelerationAltitudeDefault(): boolean {
+        return this.missedEngineOutAccelerationAltitudeDb !== undefined;
+    }
+
+    /**
+     * Finds the lowest climb constraint in the flight plan
+     * @returns the lowest climb constraint in feet or Infinity if none
+     */
+    private lowestClimbConstraint(): number {
+        let lowestClimbConstraint = Infinity;
+        for (const wp of this.waypoints) {
+            const climbConstraint = wp.additionalData.constraintType === WaypointConstraintType.CLB ? ManagedFlightPlan.climbConstraint(wp) : Infinity;
+            if (climbConstraint < lowestClimbConstraint) {
+                lowestClimbConstraint = climbConstraint;
+            }
+        }
+        return lowestClimbConstraint;
+    }
+
+    /**
+     * Check if the thrust reduction altitude is limited by a constraint and reduce it if so
+     * @returns true if a reduction occured
+     */
+    public reconcileThrustReductionWithConstraints(): boolean {
+        const lowestClimbConstraint = this.lowestClimbConstraint();
+        if (isFinite(lowestClimbConstraint) && this.thrustReductionAltitude > lowestClimbConstraint) {
+            this.thrustReductionAltitudeDb = this.thrustReductionAltitudeDb !== undefined ? Math.min(this.thrustReductionAltitudeDb, lowestClimbConstraint) : undefined;
+            this.thrustReductionAltitudePilot = this.thrustReductionAltitudePilot !== undefined ? Math.min(this.thrustReductionAltitudePilot, lowestClimbConstraint) : undefined;
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if the acceleration altitude is limited by a constraint and reduce it if so
+     * @returns true if a reduction occured
+     */
+    public reconcileAccelerationWithConstraints(): boolean {
+        const lowestClimbConstraint = this.lowestClimbConstraint();
+        if (isFinite(lowestClimbConstraint) && this.accelerationAltitude > lowestClimbConstraint) {
+            this.accelerationAltitudeDb = this.accelerationAltitudeDb !== undefined ? Math.min(this.accelerationAltitudeDb, lowestClimbConstraint) : undefined;
```

Comment:
I could introduce a third altitude derived from constraints I guess...

Reply:
Perhaps this would be a bit overkill. I am fine with keeping it like this. Another option would be to rename it from `...Db` to something along the lines of `...Default` but I believe you have used that term already.

Reply:
I thought about this a bit, and actually I don't think it should go back to the higher altitude if you delete the constraint, so this code is correct. The scratchpad message to advise the change is only specified to happen when a climb constraint is lower => the altitudes are lowered.